src/test/java/apoc/custom/CypherProceduresClusterTest.java

```diff
@@ -151,4 +151,38 @@ public void shouldRemoveFunctionOnOtherClusterMembers() throws InterruptedExcept
             throw e;
         }
     }
+
+    @Test
+    public void overrideCallStatementWithNewArguments() throws Exception {
+        // given
+        // adding 2 procedures
+        cluster.getSession().run("call apoc.custom.asProcedure('answer','RETURN 42 as answer', 'read', [['answer', 'number']])");
+        cluster.getSession().run("call apoc.custom.asProcedure('answer2','RETURN 32 as answer', 'read', [['answer', 'number']])");
+        testCallInReadTransaction(cluster.getSession(), "call custom.answer() yield answer return answer", (row) -> assertEquals(42L, row.get("answer")));
+        testCallInReadTransaction(cluster.getSession(), "call custom.answer2() yield answer return answer", (row) -> assertEquals(32L, row.get("answer")));
+
+        // when
+        // removing one procedure
+        cluster.getSession().run("call apoc.custom.removeProcedure('answer')");
+        try {
+            Thread.sleep(1000);
+            Assert.fail("procedure not removed");
+        } catch (Exception e) {
+            assertTrue(e instanceof DatabaseException);
+        }
+        // creating a new one, this triggers the scheduler
+        cluster.getSession().run("call apoc.custom.asProcedure('answer','RETURN $input as answer','read',[['answer','number']],[['input','int','42']], 'Procedure that answer to the Ultimate Question of Life, the Universe, and Everything')");
+        // waiting that the scheduler has been triggered
+        Thread.sleep(1000);
```

Comment:
could you call `apoc.custom.list` until the procedure appears (or a timeout is reached) instead?

Reply:
done, I also added a test for the `apoc.custom.list`

Reply:
hmmm, I might be missing something, but my thinking was we could replace `Thread.sleep(1000)`
with sth like (warning: not tested at all):
```java
boolean inList = false;
while (!inList) { // TBD: timeout, to avoid running an infinite loop in case of issues
       // TBD: result from `CALL apoc.custom.list()`
       inList = Iterators.asList(result)
                         .stream()
                         .map(row -> row.get("name"))
                         .filter(name -> "answer".equals(name))
                         .findAny()
                         .isPresent();
}
```

Reply:
Sorry can you elaborate why we should do that? The `Thread.sleep(1000);` is in there just to be sure that the scheduler has been executed and a cycle of `restoreProcedures` is completed, is not connected to the `apoc.custom.list`:

https://github.com/neo4j-contrib/neo4j-apoc-procedures/blob/692db025d13a0eb7b488e3485aa6223c4b315740/src/main/java/apoc/custom/CypherProcedures.java#L597

Reply:
It seems to me that `Thread.sleep(1000)` relies on a "magic value", based on the internal knowledge of the current procedure/function registration implementation. Is it guaranteed that waiting 1 second is always to be enough under any circumstance?

Taking a step back, I was thinking of the following invariants:

- a procedure/function is registered as soon as it is listed (with `apoc.custom.list` e.g.)
- a procedure/function is removed as soon as it is not listed (with `apoc.custom.list` e.g.)

Implementing the wait loop based on these seem a little more future-proof?

Reply:
Ok I did it can you please check?

Reply:
My previous comment was an attempt to explain why I suggested the (incomplete) code snippet above :)

Reply:
I used `assertEventually` that should basically do the same, do you prefer something else?